packages/shared/enqueueTask.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import warningWithoutStack from './warningWithoutStack';
+
+let didWarnAboutMessageChannel = false;
+let enqueueTask;
+try {
+  // assuming we're in node, let's try to get node's
+  // version of setImmediate, bypassing fake timers if any
+  let r = require; // trick packagers not to bundle this stuff.
+  enqueueTask = r('timers').setImmediate;
```

Comment:
Do we know for sure webpack won't detect this / warn about this?
Especially if this turns into a `require` call with a GCC pass.

Reply:
good point. I'll give it a spin later today/tomorrow

Reply:
Moment.js is using [the same method](https://github.com/moment/moment/blob/develop/src/lib/locale/locales.js#L55-L56) and **SPOILER ALERT** it doesn't work:

<img width="353" alt="image" src="https://user-images.githubusercontent.com/3297808/55454901-4acd5b00-562d-11e9-876a-bcf220fa640b.png">

Reply:
I see. I have an idea to work around it, will try later today. But how do people use moment with webpack then?

Reply:
The _actual_ thing you need is `__non_webpack_require__`

Reply:
Aha, I see the problem, people futz around with plugins to exclude locales. Ok cool, i’ll try a workaround. Thanks for the heads up!

Reply:
We might not be able to special case just for webpack. I remember trying this a while ago and it working `(() => eval(‘require’))`

Reply:
To be clear, this is also a bit of an edge case, you’d have to be using webpack on your tests for this to be an issue. But will try to fix this before releasing.

Reply:
(Great, the new guy gets to ship eval into react. Just great. 🍿)

Reply:
Another option, if one’s running their tests through webpack, would be to mark react/react-dom as externals.

Reply:
fwiw that's how I outsmarted the compiler once:

```
export function requireAtRuntime(moduleName: string): any {
  const global = getGlobal()
  const require = global[atob('cmVxdWlyZQ==')]
  return require(moduleName)
}
```

Reply:
We definitely shouldn't be adding special hacks for webpack alone.

Reply:
Eval also won't work because it's restricted in some environments.

Reply:
>if one’s running their tests through webpack

Does this have something to do with tests? I thought this file would be pulled in by all renderers. Regardless of whether you're writing tests or not.

Reply:
Oh I see, react-dom doesn't pull it in.

Reply:
I checked CRA, and seems like this isn't a problem there. Maybe it's ok to wait until somebody actually reports it.

Reply:
In CRA this causes webpack's polyfill of `timers` to get bundled.

Reply:
Could enable jest.useFakeTimers() and await an async act() call to see if it proceeds?

Reply:
I don't understand what you mean

Reply:
https://github.com/facebook/react/pull/15305